tc/sgx/trusted_worker_manager/enclave_wrapper/file_io_processor.cpp

```diff
@@ -0,0 +1,298 @@
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            return FAILED;
+        }
+
+        // write buffer data to file
+        fileStream.write((const char*)inBuf, inBufSize);
+        tcf::Log(TCF_LOG_INFO, "File %s WRITE has been completed\n", fileName.c_str());
+        fileStream.close();
+
+    } catch (std::ofstream::failure e) {
+        tcf::Log(TCF_LOG_ERROR, "Caught exception while Writing file: %s\n", e.what());
+        resultStr = "FAILED TO WRITE: Caught Exception while writing file";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    resultStr = "FILE WRITE SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Tells current position of the file pointer and stores it in buffer
+// updates status SUCCESS or FAILED in result parameter
+uint32_t FileTell(std::string fileName, uint8_t *result, size_t resultSize,
+    uint8_t *outBuf, size_t outBufSize) {
+    string resultStr;
+    if (fileName.empty()) {
+        tcf::Log(TCF_LOG_ERROR, "Filename is empty\n");
+        resultStr = "FAILED TO FTELL: Filename is empty";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    try {
+        ifstream fileStream (fileName);
+        if (!fileStream.is_open()) {
+            tcf::Log(TCF_LOG_ERROR, "Unable to open file %s\n", fileName.c_str());
+            resultStr = "FAILED TO FTELL: Unable to open file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+
+        size_t index = fileStream.tellg();
+
+        tcf::Log(TCF_LOG_INFO, "File %s ftell has been completed\n", fileName.c_str());
+        outBuf = (uint8_t*) index;
+        outBufSize = sizeof(index);
+        fileStream.close();
+
+    } catch (std::ifstream::failure e) {
+        tcf::Log(TCF_LOG_ERROR, "Caught exception while ftell file: %s\n", e.what());
+        resultStr = "FAILED TO FTELL: Caught Exception while doing ftell";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    resultStr = "FILE FTELL SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Places file pointer to the specific position updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileSeek(std::string fileName, size_t position, uint8_t *result,
+    size_t resultSize) {
```

Comment:
What is the purpose of the `resultSize` parameter? It is unused. The incoming parameter value is always overwritten in the function and the incoming value is never used.
You should test that the string copied to `result` never exceeds the input value of `resultSize` (including the `NULL` character).
Also document in in the function comments that `resultSize` is the maximum size of the `result` buffer.

Reply:
I agree, we need to use resultSize as max threshold value of the string value that can be copied to result parameter. 
I will add this change along with documentation.